### ISSUE TITLE
OpenSSL Vulnerability updates, CVE-2015-0291 and others

### ISSUE THREAD

**Issue by yosifkit** [docker-library/official-images#569]
Specifically the following vulnerabilities announced and fixed today (from [openssl.org](https://www.openssl.org/news/vulnerabilities.html)):

- CVE-2015-0291 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0291), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0291.html))
- CVE-2015-0290 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0290), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0290.html))
- CVE-2015-0207 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0207), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0207.html))
- CVE-2015-0286 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0286), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0286.html))
- CVE-2015-0208 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0208), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0208.html))
- CVE-2015-0287 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0287), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0287.html))
- CVE-2015-0289 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0289), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0289.html))
- CVE-2015-0292 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0292), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0292.html))
- CVE-2015-0293 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0293), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0293.html))
- CVE-2015-1787 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-1787), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-1787.html))
- CVE-2015-0209 ([Debian](https://security-tracker.debian.org/tracker/CVE-2015-0209), [Ubuntu](http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0209.html))

**Comment by yosifkit** [comment]
Fixes for debian (the others not listed here already have fixes):
https://security-tracker.debian.org/tracker/CVE-2015-0286
https://security-tracker.debian.org/tracker/CVE-2015-0287
https://security-tracker.debian.org/tracker/CVE-2015-0289
https://security-tracker.debian.org/tracker/CVE-2015-0292
https://security-tracker.debian.org/tracker/CVE-2015-0209

**Comment by tianon** [comment]
Updated OP with all Debian and Ubuntu tracker links.

**Comment by tianon** [comment]
Ubuntu has a slightly longer list:
http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0286.html
http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0287.html
http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0289.html
http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0292.html
http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0293.html
http://people.canonical.com/~ubuntu-security/cve/2015/CVE-2015-0209.html

**Comment by ewindisch** [comment]
Debian is carrying the package in wheezy-security repo. The debian image is not affected directly, but dependent images still need updates as they 'apt-get update; apt-get install'... this includes buildpack-deps

**Comment by ewindisch** [comment]
The following RHEL page generally will apply to the centos and oraclelinux images: https://access.redhat.com/articles/1384453

**Comment by ewindisch** [comment]
Use this to track Ubuntu progress: https://launchpad.net/ubuntu/+source/openssl

Packages currently in -proposed repo

**Comment by ewindisch** [comment]
Upstream packages available
- [x] debian
- [x] buildpack-deps

Updated or no update necessary
- [x] debian (no openssl package)
- [ ] buildpack-deps
- [ ] fedora
- [ ] centos
- [ ] oraclelinux
- [ ] mono (builds on debian:wheezy... does mono use openssl?  /cc @directhex)

**Comment by directhex** [comment]
No, Mono doesn't use OpenSSL & I don't think it's part of our image. We have our own TLS stack with our own fresh CVEs for you, though - see https://github.com/docker-library/official-images/pull/570

**Comment by ewindisch** [comment]
thanks @directhex

**Comment by tianon** [comment]
From my reading, only wheezy is updated.  We need those updates spread
across the other suites too (minimally sid).

**Comment by ewindisch** [comment]
@tianon noted. I can mark that on my tracking comment. We still need  to update the dependent official images which depend on wheezy.

**Comment by ewindisch** [comment]
@tianon correction - Debian has since updated sid and other suites.

**Comment by tianon** [comment]
Security tracker isn't showing that sid or jessie are fixed; where are you seeing that?

**Comment by ewindisch** [comment]
I was looking at:
https://security-tracker.debian.org/tracker/CVE-2015-0293

Most of the affected images are running wheezy and we should prioritize
updating those images.

(I'd also argue we should not have any official images running jessie/sid
and this should be filed as a new issue)
ᐧ

**Comment by tianon** [comment]
As a status update, all images except for the following have been re-pushed: (and these are still in progress; we've had some hiccups due to some Hub issues like the SSL cert expiring right in the middle of our pushing)
```
tomcat:8-jre8
tomcat:jre8
ubuntu-upstart:12.04
ubuntu-upstart:precise
ubuntu-upstart:14.04
ubuntu-upstart:trusty
ubuntu-upstart:latest
ubuntu-upstart:14.10
ubuntu-upstart:utopic
websphere-liberty:8.5.5
websphere-liberty:latest
websphere-liberty:beta
wordpress:4.1.1-apache
wordpress:4.1.1
wordpress:4.1-apache
wordpress:4.1
wordpress:4-apache
wordpress:apache
wordpress:4
wordpress:latest
wordpress:4.1.1-fpm
wordpress:4.1-fpm
wordpress:4-fpm
wordpress:fpm
```

**Comment by tianon** [comment]
Ok, images are finally finished pushing. :+1: (and Jessie is _almost_ updated upstream, too https://qa.debian.org/excuses.php?package=openssl)

**Comment by tianon** [comment]
https://bugs.debian.org/780902 has the latest for why that isn't unblocked completely yet (there's a new debian-installer RC due this week, after which it'll be unblocked and transition to jessie)

**Comment by tianon** [comment]
`oraclelinux` is now updated (as of #618)

**Comment by ewindisch** [comment]
@tianon can we update the list I made earlier? I'm not sure if you can edit it directly...

**Comment by tianon** [comment]
Yep, just updated a bunch of it.

**Comment by ewindisch** [comment]
@tianon I see you noted mysql and percona. Did this include updating the mysqlclient libraries? These are statically linked to openssl.

**Comment by tianon** [comment]
Are you sure?  I marked them because they're both "FROM debian:wheezy",
which has been updated, and they're both deb-package-based, not pulling
from binaries.

**Comment by tianon** [comment]
Fix for jessie just dropped upstream.  I'm rebuilding images (and just doing all the Debian suites again for good measure and consistency).

**Comment by tianon** [comment]
Updated base images are pushed and updated dependent images are built; all that's left is the pushing (which is currently in-progress).
